src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java

```diff
@@ -1216,13 +1216,21 @@ public int nextCycle(int currentCycle, @NotNull TailerDirection direction) {
                 fileFound |= currentCycleFile.exists();
 
                 if (!fileFound) {
-                    directoryListing.refresh(true);
-                    throw new MissingStoreFileException(
-                            String.format("Expected file to exist for cycle: %d, file: %s.%nminCycle: %d, maxCycle: %d%n" +
-                                            "Available files: %s",
-                                    currentCycle, currentCycleFile,
-                                    directoryListing.getMinCreatedCycle(), directoryListing.getMaxCreatedCycle(),
-                                    Arrays.toString(path.list((d, n) -> n.endsWith(SingleChronicleQueue.SUFFIX)))));
+                    Jvm.error().on(SingleChronicleQueue.class, "The current cycle seems to have been deleted from under the queue, scanning to find the next remaining cycle");
```

Comment:
May lead to data loss going unnoticed by our users...

Reply:
Maybe we should make this behaviour configurable?

Reply:
https://www.joelonsoftware.com/2000/04/12/choices/

Reply:
That's a change of behavior and a quite significant one at that.

Reply:
Yes,  but I think we should either decide to do it or not, I'm not a fan of adding another config option. It just adds to an already large config space meaning another configuration that isn't comprehensively tested.

Reply:
I would prefer to not do it, because users can work around failures but they cannot work around silent data loss caused by suppression of failures.
But that's just my opinion.

Reply:
A couple of things about the change
- It does log an error to the logs when it goes into this recovery mode, so not completely "silent"/"suppressed"
- This recovery only occurs when you delete the current AND next roll cycle, if you delete EITHER the current or the next cycle the existing code will silently continue, so it's not like there's any protection against data loss in the existing behaviour that's being removed
- If you don't want to lose data you probably shouldn't be deleting files from under the program